dynamic-datasource-spring-boot-starter/src/test/java/com/baomidou/dynamic/datasource/fixture/v1/AddRemoveDatasourceTest.java

```diff
@@ -53,6 +53,13 @@ void testAddAndRemoveDataSource() {
         dataSourceProperty.setUrl("jdbc:h2:mem:test1");
         dataSourceProperty.setDriverClassName("org.h2.Driver");
         DynamicRoutingDataSource ds = (DynamicRoutingDataSource) dataSource;
+        // async destroy datasource
+        ds.setGraceDestroy(true);
+        ds.addDataSource(dataSourceProperty.getPoolName(), dataSourceCreator.createDataSource(dataSourceProperty));
+        assertThat(ds.getDataSources().keySet()).contains("slave_1");
+        ds.removeDataSource("slave_1");
+        // close directly
+        ds.setGraceDestroy(false);
```

Comment:
- The code here is missing assertions.  And I recommend making assertions in separate unit tests rather than complicating existing unit tests.